src/contract_wrappers/token_registry_wrapper.ts

```diff
@@ -20,20 +21,30 @@ export class TokenRegistryWrapper extends ContractWrapper {
         const tokenRegistryContract = await this._getTokenRegistryContractAsync();
 
         const addresses = await tokenRegistryContract.getTokenAddresses.call();
-        const tokenMetadataPromises: Array<Promise<TokenMetadata>> = _.map(
+        const tokenPromises: Array<Promise<Token|undefined>> = _.map(
             addresses,
-            (address: string) => (tokenRegistryContract.getTokenMetaData.call(address)),
+            (address: string) => (this.getTokenMetadataIfExistsAsync(address)),
         );
-        const tokensMetadata = await Promise.all(tokenMetadataPromises);
-        const tokens = _.map(tokensMetadata, metadata => {
-            return {
-                address: metadata[0],
-                name: metadata[1],
-                symbol: metadata[2],
-                decimals: metadata[3].toNumber(),
-            };
-        });
-        return tokens;
+        const tokens = await Promise.all(tokenPromises);
+        return tokens as Token[];
+    }
+    /**
+     * Retrieves a token by address currently listed in the Token Registry smart contract
```

Comment:
But it actually returns a token... I think I need to change the name of the method. Will do that now.